swarm/network/protocol.go

```diff
@@ -217,23 +214,23 @@ func (b *Bzz) performHandshake(p *protocols.Peer, handshake *HandshakeMsg) error
 		return err
 	}
 	handshake.peerAddr = rsh.(*HandshakeMsg).Addr
-	sanitizeEnodeRemote(p.RemoteAddr(), handshake.peerAddr)
+	sanitizeEnodeRemote(p.Node(), handshake.peerAddr)
 	handshake.LightNode = rsh.(*HandshakeMsg).LightNode
 	return nil
 }
 
-// the remote enode string may advertise arbitrary host information (e.g. localhost)
-// this method ensures that the addr of the peer will be the one
-// applicable on the interface the connection came in on
+// if started without the natip argument, the enode string will be localhost
+// this method ensures that if this default is used in a networked environment, we replace
+// the ip with the one applicable on the interface the connection came in on
 // it modifies the passed bzzaddr in place, and returns the same pointer
-func sanitizeEnodeRemote(paddr net.Addr, baddr *BzzAddr) {
-	hsSubmatch := regexpEnodeIP.FindSubmatch(baddr.UAddr)
-	ip, _, err := net.SplitHostPort(paddr.String())
-	// since we expect nothing else than ipv4 here, a panic on missing submatch is desired
-	if err == nil && string(hsSubmatch[1]) != ip {
-		remoteStr := fmt.Sprintf("@%s:%s", ip, string(hsSubmatch[2]))
-		log.Debug("rewrote peer uaddr host/port", "addr", baddr)
-		baddr.UAddr = regexpEnodeIP.ReplaceAll(baddr.UAddr, []byte(remoteStr))
+func sanitizeEnodeRemote(paddr *enode.Node, baddr *BzzAddr) {
```

Comment:
This function is modifying `baddr` - probably it'd be better if it is a method on `BzzAddr`.

A function with a side-effect for the second argument is somewhat unusual.